### PR TITLE
Cpp warnings

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
Adjusted some CPP Warnings #1264 
Simplified logic for EthGetworkClient
- removed unused variables
- unified behavior (startworking only after connect)

### CODE REVIEW

**Comment by AndreaLanfranchi** [comment]
At the moment yes.
Leave the branch open to push further fixes.

**Comment by XhmikosR** [comment]
Did you manage to make sense of the `%u` specifier? It seems the struct is defined with `int`s but the local variables are `unsigned`. Would be nice to streamline this.

**Comment by chfast** [comment]
Right, but using atomic with relaxed memory order will be implemented as a regular LOAD. I'm not sure if the STORE is atomic.

**Comment by jean-m-cyr** [comment]
Both loads and stores compile down to a single.

int main()
{
00101000  push        ebp  
00101001  mov         ebp,esp  
00101003  and         esp,0FFFFFFF8h  
	static std::atomic<unsigned> u;
	_u.store(10, std::memory_order_relaxed);_
00101006  **mov         dword ptr [u (0103390h)],0Ah**  
	return _u.load(std::memory_order_relaxed);_
00101010  **mov         eax,dword ptr [u (0103390h)]**  
}
00101015  mov         esp,ebp  
00101017  pop         ebp  
00101018  ret

**Comment by AndreaLanfranchi** [comment]
> We all realize that in x86 architecture, loads and stores of simple types are naturally atomic, and that the atomic designation is not required unless we need to use read-modify-write (exchange) operations

I must admit I knew that for C# but was not sure for C++.
Does the atomic designation causes unnecessary overhead ?

**Comment by jean-m-cyr** [comment]
No overhead at all.

**Comment by chfast** [comment]
I prefer C++ correctness. But moreover, one practical argument for atomics here is that we can use thread sanitizer to detect race conditions in other places.

**Comment by jean-m-cyr** [comment]
I'm not arguing against, just making the observation that it makes no practical difference in X86. I'm curious how the thread sanitizer could even detect the difference since a store to an atomic generates exactly the same code as a direct assignment?

**Comment by chfast** [comment]
Thread Sanitizer instruments the binary but on C++ level, so it sees atomic load and nonatomic load.

**Comment by jean-m-cyr** [comment]
The thread sanitizer is involved at compile time? Ok then.

File: libpoolprotocols/getwork/EthGetworkClient.cpp [hunk 1]
```diff
@@ -25,22 +24,25 @@ EthGetworkClient::~EthGetworkClient()
 
 void EthGetworkClient::connect()
 {
-	if (m_connection_changed) {
-		stringstream ss;
-		ss <<  "http://" + m_conn->Host() << ':' << m_conn->Port();
-		if (m_conn->Path().length())
-			ss << m_conn->Path();
-		p_client = new ::JsonrpcGetwork(new jsonrpc::HttpClient(ss.str()));
+    stringstream ss;
```
Comment (jean-m-cyr) [inline libpoolprotocols/getwork/EthGetworkClient.cpp]: Why use a stringstream when a string would do the job?

Use + or += to concatenate strings, and to_string(n) to convert the port #.

Comment (AndreaLanfranchi) [reply libpoolprotocols/getwork/EthGetworkClient.cpp]: It's old legacy code. 
Will amend.

**Pull request merged**